testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/config/smallrye/management/config_source/from_root_dir/ConfigSourceRootTestCase.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir;
+
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.X_D_OVERRIDES_A;
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.FROM_A1;
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.FROM_A2;
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.FROM_B;
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.Y_A_OVERRIDES_B;
+import static org.wildfly.test.integration.microprofile.config.smallrye.management.config_source.from_root_dir.TestApplication.Z_C_OVERRIDES_A;
+
+import java.net.URL;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.microprofile.config.smallrye.AbstractMicroProfileConfigTestCase;
+import org.wildfly.test.integration.microprofile.config.smallrye.AssertUtils;
+
+/**
+ * Tests Root directory config sources
+ *
+ * @author Kabir Khan
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(SetupTask.class)
+public class ConfigSourceRootTestCase extends AbstractMicroProfileConfigTestCase {
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "ConfigSourceFromDirTestCase.war")
+                .addClasses(TestApplication.class, TestApplication.Resource.class, AbstractMicroProfileConfigTestCase.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return war;
+    }
+
+    @ArquillianResource
+    private URL url;
+
+    @Test
+    public void testGetWithConfigProperties() throws Exception {
```

Comment:
So my understanding is this is only test for this feature. What I am going to do tomorrow is to understand if all hard requirements [1] are covered by this test. But I see 9 points in hard requirements, whereas I see 6 tests here. So probably something will be missing?

Maybe some comments could help me understand which test is testing which use case.

[1] https://github.com/wildfly/wildfly-proposals/pull/450/files#diff-9f1354c78479d2107750d078d4f846420ac82216e6eda99d34a0ad36ea2d948fR66

Reply:
1. Do we have test that root=false is default? Do we have test that if root=false is configured, resolving of nested directories is not working?
Basically I am looking for covering this hard requirement `The default value of the `root` nested attribute is `false`, giving the legacy behaviour`
2. I do not see nested dirs in setup task so I believe those requirements are not tested:
Basically I am looking for covering this hard requirement  `Only top-level directories are considered, directories nested within those are ignored`
3. I do not see files created in root directory
Basically I am looking for covering this hard requirement `Files contained directly within the root directory are ignored and are not used for config`

Reply:
@mchoma I believe I have added what you are asking for now (good catch)

I've addressed 1) in ConfigSourceFromDirTestCase. It now adds one of the config sources with the default value for `root` and the other explicitly as `root=false`. In both directories I create a nested directory containing a property file each, and verify those properties are not considered.

For 2 + 3) I have modified ConfigSourceRootTestCase. 
For 2) I added a file in a directory nested under one of the directories under the root dir.
For 3) I added a file directly under the root directory. 
Neither of these files get considered as properties.